### ISSUE TITLE
Reproduce RCV1 score

### ISSUE THREAD

**Issue by ayushbits** [bcol23/HyperIM#5]
Hey @bcol23 ,

Thanks for the paper and the code.
I adopted following steps to reproduce your numbers :

1. Pre-train the 300d word embeddings from poincare_glove with the following command
`./run_glove.sh --train --coocc_file ../poincare_glove2/GloVe/cooccurrence.bin --vocab_file ../poincare_glove2/GloVe/vocab.txt --epochs 50 --workers 100 --restrict_vocab 200000 --lr 0.01 --bias --dist_func cosh-dist-sq --root .. --chunksize 1000 --poincare 1 --no_eval --mix --num_embs 50 --size 300`
I couldn't understand the initialization part mentioned in Issue #1. 

2. Pre-train the 10 dim label embeddings from poincare embeddings.
3. I replaced the word and label embeddings in HyperIM.py file and ran the model to recieve the following scores for RCV1.
P@1     55.369          P@3     36.741          P@5     28.032
r@1     44.631          r@3     63.259          r@5     71.968
MicroF1@1       45.817          MicroF1@3       45.845          MicroF1@5       39.947
nDCG@1  55.369          nDCG@3  44.331          nDCG@5  46.696

These numbers are far less than reported in the paper. 
Could you tell what additional needs to be done to achieve scores mentioned in your paper

Thanks,

**Comment by bcol23** [comment]
To reproduce the results of *HyperIM* in Table 2, first change the interaction dimension to 75x2D (*i.e.*, a Cartesian product of 75 2D Poincare balls). The corresponding results of *EuclideanIM* should also be checked. It is recommended to first test the *EuclideanIM* and *HyperIM* with random initialized embeddings.

**Comment by bcol23** [comment]
To reproduce the results of HyperIM in Table 2, first change the interaction dimension to 75x2D (i.e., a Cartesian product of 75 2D Poincare balls). The corresponding results of EuclideanIM should also be checked. It is recommended to test the EuclideanIM and HyperIM with random initialized embeddings.
